### ISSUE TITLE
Thunderbird 102 - folder tree icons do not work independently of tabs

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#280]
User-defined Icons that are defined for a folder in the tree without a matching Folder tab are not retrieved / displayed anymore - at least in large profiles converted from older versions of Thunderbird. We may also see many warnings of exceptions caught in folder tree, similar to this:

```
QuickFolders.FolderTree.getCellPropertiesQuickFolders.FolderTree.getCellProperties(14,[object TreeColumn])
  returning unchanged props for folder [nstmp] : 
Component returned failure code: 0x80550007 [nsIMsgFolder.getStringProperty]
```
which may be a side effect of invalid icons (to be investigated). With many messages like this, performance may be impacted adversely.

It looks like we will need to store custom icon settings externally (so far I managed to store them in the folders themselves by monkey-patching the function `gFolderTreeView.getCellProperties` - if we store icon definitions in QuickFolders extension storage (which gets stored in prefs.js) it will be necessary to redefine all icons for (non-tab) folders that are not declared via a dedicated QuickFolders tab.

**Comment by RealRaven2000** [comment]
In order to make testing a little easier and improve performance + signal-to-noise ratio in the error log, here is a version of QuickFolders  that omits the message (the error is still thrown, but it happens "quietly":

[QuickFolders-mx-5.10.1pre7.zip](https://github.com/RealRaven2000/QuickFolders/files/9033396/QuickFolders-mx-5.10.1pre7.zip)

To install this version, download the zip file and drag into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
Fixes most issues for icons in folder tree:

[QuickFolders-mx-5.11pre85.zip](https://github.com/RealRaven2000/QuickFolders/files/9127872/QuickFolders-mx-5.11pre85.zip)

To install this version, download the zip file and drag into Thunderbird Add-ons Manager.

**Comment by RealRaven2000** [comment]
I removed an unnecessary call to the function `repairTreeIcons()` on startup to save on processing time in Thunderbird 102 - this will also have additional effects for issue #283.

**Comment by RealRaven2000** [comment]
Fixed in 5.11 - Published 23/07/2022
